Kitodo-FileManagement/src/main/java/org/kitodo/filemanagement/FileManagement.java

```diff
@@ -179,6 +183,21 @@ public URI rename(URI uri, String newName) throws IOException {
         return performRename(mappedFileURI, mappedNewFileURI);
     }
 
+    private URI renameDirectory(URI directoryPath, String newDirectoryName) throws IOException {
+        String uriString = directoryPath.toString();
+        if (uriString.equals(newDirectoryName + "/") || uriString.equals(newDirectoryName)) {
+            return null;
+        }
+        File newDirectory = new File(KitodoConfig.getKitodoDataDirectory() + newDirectoryName);
+        File oldDirectory = new File(KitodoConfig.getKitodoDataDirectory() + directoryPath);
+        boolean success = oldDirectory.renameTo(newDirectory);
+        if (success) {
+            return newDirectory.toURI();
+        } else {
+            throw new IOException("Unable to rename directory '" + directoryPath + "'!");
+        }
+    }
```

Comment:
Maybe in this way:

```java
    private URI renameDirectory(URI directoryPath, String newDirectoryName) throws IOException {
        File newDirectory = new File(KitodoConfig.getKitodoDataDirectory() + newDirectoryName);
        File oldDirectory = new File(KitodoConfig.getKitodoDataDirectory() + directoryPath);

        if (! oldDirectory.equals(newDirectory)) {
            boolean success = oldDirectory.renameTo(newDirectory);
            if (success) {
                return newDirectory.toURI();
            } else {
                throw new IOException("Unable to rename directory '" + directoryPath + "'!");
            }
        } else {
            return newDirectory.toURI();
        }
    }
```

Returning `null` should be avoided and the changed JavaDoc comment could be changed to `URI of renamed resource`